presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/RewriteAggregationIfToFilter.java

```diff
@@ -228,6 +239,12 @@ private boolean shouldRewriteAggregation(Aggregation aggregation, ProjectNode so
             return false;
         }
         RowExpression sourceExpression = sourceProject.getAssignments().get((VariableReferenceExpression) aggregation.getArguments().get(0));
+        if (sourceExpression instanceof CallExpression
+                && ((CallExpression) sourceExpression).getArguments().size() == 1
+                && "CAST".equals(((CallExpression) sourceExpression).getDisplayName())) {
```

Comment:
Store `FunctionResolution` as a final member variable and use `functionResolution.isCastFunction(callExpression.getFunctionHandle())`